runtime/lua/vim/net.lua

```diff
@@ -0,0 +1,533 @@
+local M = {}
+
+---@class HeaderTable
+---@field private _storage table<string, string[]>
+local HeaderTable = {}
```

Comment:
This looks a lot like a [multimap](https://en.m.wikipedia.org/wiki/Multimap).  Maybe worth having this as `vim.multimap` or `vim.multidict`, similar to the defaultdict - instead of exposing a specialised HeaderTable

Reply:
the requirements are a multimap data structure with case-insensitive operations, either vim.multimap includes that as an option or the HeaderTable stays as a wrapper.